chromium_src/content/browser/worker_host/shared_worker_host.cc

```diff
@@ -0,0 +1,28 @@
+/* Copyright (c) 2021 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include "src/content/browser/worker_host/shared_worker_host.cc"
+
+namespace content {
+
+void SharedWorkerHost::AllowFingerprinting(
+    const GURL& url,
+    base::OnceCallback<void(bool)> callback) {
+  std::move(callback).Run(
+      GetContentClient()->browser()->AllowWorkerFingerprinting(
+          url, GetProcessHost()->GetBrowserContext(),
+          GetRenderFrameIDsForWorker()));
+}
+
+void SharedWorkerHost::GetBraveFarblingLevel(
+    const GURL& url,
+    base::OnceCallback<void(uint8_t)> callback) {
+  std::move(callback).Run(
```

Comment:
I'm not sure, but do you really need callbacks here? I think you can just return the value directly and run a callback in the `Impl` part directly.

Reply:
This is how upstream does things for similarly proxied methods like `AllowWebLocks`, `AllowIndexedDB`, &c. I assume it's required somehow by the generated mojo classes. Anyway I'm not changing it.